R/guides.R

```diff
@@ -59,13 +59,13 @@ collapse_guides <- function(guides) {
   }
   guides
 }
+
 #' @importFrom gtable gtable_width gtable_height gtable gtable_add_grob
 #' @importFrom grid editGrob heightDetails widthDetails valid.just unit.c unit
 #' @importFrom ggplot2 margin element_grob element_blank calc_element element_render
 guides_build <- function(guides, theme) {
-  theme$legend.spacing <- calc_element("legend.spacing", theme) %||% unit(0.5, "lines")
-  legend.spacing.y <- calc_element("legend.spacing.y", theme)
-  legend.spacing.x <- calc_element("legend.spacing.x", theme)
+  legend.spacing.y <- .subset2(theme, "legend.spacing.y")
+  legend.spacing.x <- .subset2(theme, "legend.spacing.x")
```

Comment:
What is the rationale for this change? I would expect this to break if the legend spacing is provided through inheritance in the theme...